nomad/job_endpoint_hook_consul.go

```diff
@@ -30,3 +30,63 @@ func (jobConsulHook) validateTaskPartitionMatchesGroup(groupPartition string, ta
 	}
 	return nil
 }
+
+// mutateImpl ensures that the job's Consul blocks have been configured with the
+// correct Consul cluster if unset, and sets constraints on the Consul admin
+// partition if set. This should be called by the Mutate method.
+func (jobConsulHook) mutateImpl(job *structs.Job, defaultCluster string) (*structs.Job, []error, error) {
+	for _, group := range job.TaskGroups {
+		if group.Consul != nil {
+			if group.Consul.Cluster == "" {
+				group.Consul.Cluster = defaultCluster
+			}
+			if group.Consul.Partition != "" {
+				group.Constraints = append(group.Constraints,
+					newConsulPartitionConstraint(group.Consul.Cluster, group.Consul.Partition))
+			}
+		}
+
+		for _, service := range group.Services {
+			if service.IsConsul() && service.Cluster == "" {
+				service.Cluster = defaultCluster
+			}
+		}
+
+		for _, task := range group.Tasks {
+			if task.Consul != nil {
+				if task.Consul.Cluster == "" {
+					task.Consul.Cluster = defaultCluster
+				}
+				if task.Consul.Partition != "" {
+					task.Constraints = append(task.Constraints,
+						newConsulPartitionConstraint(task.Consul.Cluster, task.Consul.Partition))
+				}
+			}
+			for _, service := range task.Services {
+				if service.IsConsul() && service.Cluster == "" {
+					service.Cluster = defaultCluster
+				}
+			}
+		}
+	}
+
+	return job, nil, nil
```

Comment:
Feels a little strange for the function signature to include []error and error, but for there to be no condition where they're not nil.  The `Mutate()` function needs to, to implement the interface, but here it strikes me as odd.

Reply:
Eh, good point. The ENT version potentially returns errors but it can do that in the call. Will fix.